### ISSUE TITLE
New Changes in constructor

### ISSUE THREAD

**Issue by Ma5t3rful** [mmertama/Gempyre#12]
Not really an issue just a suggestion

``` cpp
//Ui(const Filemap& filemap, const std::string& indexHtml, const std::string& title = "",  int width = -1, int height = -1, unsigned flags = 0, unsigned short port = UseDefaultPort, const std::string& root = UseDefaultRoot);

/// Create window UI
Ui(const Filemap& filemap, const std::string& indexHtml, const std::string& title,  int width, int height,
            const std::unordered_map<std::string, std::string>& ui_params = {}, unsigned flags = 0, unsigned short port = UseDefaultPort, const std::string& root = UseDefaultRoot);
```
@mmertama 
can we keep that constructor that is just commented out in the future? `flags` are cleaner than an unordered-map IMO. maybe just flip `flags` with `params` like this so that we can be more compatible with the constructor that you have commented out:

```cpp
/// Create window UI
Ui(const Filemap& filemap, const std::string& indexHtml, const std::string& title,  int width, int height,  unsigned flags = 0,
            const std::unordered_map<std::string, std::string>& ui_params = {}, unsigned short port = UseDefaultPort, const std::string& root = UseDefaultRoot);
```

Cheers

**Comment by mmertama** [comment]
I supposedly should add some comment but here is some explanation of the change:   The reason behind the change is that Python vs browser UI is now dynamic (instead of USE_PYTHON_UI flag) and c'tor now should reflect better what UI is preferred.

The removed c'tor: 
`Ui(const Filemap& filemap, const std::string& indexHtml, const std::string& title = "",  int width = -1, int height = -1, unsigned flags = 0, unsigned short port = UseDefaultPort, const std::string& root = UseDefaultRoot);`

As it looks like the most common use case (at least I have not seen) was to be happy with all default parameters I decided to shorten that as: 

`Ui(const Filemap& filemap, const std::string& indexHtml, unsigned short port = UseDefaultPort, const std::string& root = UseDefaultRoot);`

` - that is 'legacy' c'tor the current implementation is that its up to USE_PYTHON_UI whether that uses browser or Python UI - but I dont know if that makes any sense.  There will be something more clever in the future.

```
       Ui(const Filemap& filemap, const std::string& indexHtml, const std::string& browser,  const std::string& browser_params, unsigned short port = UseDefaultPort, const std::string& root = UseDefaultRoot);

```
This is browser c'tor , uses browser UI

```
        Ui(const Filemap& filemap, const std::string& indexHtml, const std::string& title,  int width, int height,
            const std::unordered_map<std::string, std::string>& ui_params = {}, unsigned flags = 0, unsigned short port = UseDefaultPort, const std::string& root = UseDefaultRoot);
```

This is new one, it tries to open Python UI - the rd parameter is passed to Python as well as flags and they have quite webview spesific meaning. I wrote some document in gemgui and shall copied that here.  However they have their special purpose and meaning.
